shared/node.go

```diff
@@ -31,14 +32,20 @@ func CreateNode(bus modules.Bus) (modules.Module, error) {
 }
 
 func (m *Node) Create(bus modules.Bus) (modules.Module, error) {
+	p2pConfig := bus.GetRuntimeMgr().GetConfig().P2P
+	p2pCreate := p2p.Create
+	if p2pConfig.UseLibP2P {
```

Comment:
We could do something like this in `p2p/module.go`'s `Create` function but I'm not sure how I feel about it yet. I worry it's a bit indirect and could be confusing for the reader:

```
func Create(bus modules.Bus) (modules.Module, error) {
	p2pConfig := bus.GetRuntimeMgr().GetConfig().P2P
	if p2pConfig.UseLibP2P {
		return new(p2pModule).Create(bus)
	}

	return libp2p.Create(bus)
}
```

(This would DRY up the part in `app/client/cli/debug.go` that you were commenting on as well.)

Reply:
I think the `UseLibP2P` should be part of the base config and not the P2PConfig since its a competing module.

Theoretically, you could have a `LibP2PConfig` which would be a mostly copy-pasta which is how, theoretically, competing modules could do it in the future. I think we can stick with the same config for now, but move this out.

Reply:
> I think the `UseLibP2P` should be part of the base config and not the P2PConfig since its a competing module.

Yes, thank you once again for the right context at the right time! :stuck_out_tongue_closed_eyes: 

> Theoretically, you could have a `LibP2PConfig` which would be a mostly copy-pasta which is how, theoretically, competing modules could do it in the future. I think we can stick with the same config for now, but move this out.

I had a similar thought but wasn't ready to commit yet either. Good to know we're on the same page! I'll add a comment and a mental note to do so before review in the future to better streamline these discussion. :sweat_smile: